File: ocaml/diamond.ml
class a = object method foo = print_endline "A" end
class b = object inherit a method foo = print_endline "B" end
class c = object inherit a method foo = print_endline "C" end
class d = object inherit b inherit c (* Last parent (C) wins *) end

let () = (new d)#foo (* Output: C *)

File: php/diamond.php
<?php
interface IA { public function foo(); }
interface IB extends IA { public function foo(); }
interface IC extends IA { public function foo(); }

class D implements IB, IC {
    public function foo() { echo "D"; } // Must implement
}

$d = new D();
$d->foo(); // Output: D